django/db/migrations/graph.py

```diff
@@ -156,7 +211,7 @@ def make_state(self, nodes=None, at_end=True, real_apps=None):
             nodes = list(self.leaf_nodes())
         if len(nodes) == 0:
             return ProjectState()
-        if not isinstance(nodes[0], tuple):
+        if not isinstance(nodes[0], (tuple, Node)):
```

Comment:
That worked because `node == node.key` and `hash(node) == hash(node.key)`. `leaf_nodes()` and `root_nodes()` also used to return actual nodes instead of keys. It still works, but I'm now consistently returning tuples instead of nodes in all functions, so I've removed the instance check.